src/ifcconvert/IfcConvert.cpp

```diff
@@ -589,30 +593,62 @@ int main(int argc, char** argv)
 
 	serializer->writeHeader();
 
-	int old_progress = -1;
+    if (center_model || model_offset || site_local_placement) {
+		int flag_count =
+			center_model ? 1 : 0
+			+ model_offset ? 1 : 0
+			+ site_local_placement ? 1 : 0;
+		if (flag_count > 1) {
+			Logger::Error("Can only use one of --center-model, --model-offset and --site-local-placement");
+			delete serializer;
+			return EXIT_FAILURE;
+		}
 
-    if (center_model || model_offset) {
-        double* offset = serializer->settings().offset;
+        std::stringstream msg;
         if (center_model) {
             gp_XYZ center = (context_iterator.bounds_min() + context_iterator.bounds_max()) * 0.5;
-            offset[0] = -center.X();
-            offset[1] = -center.Y();
-            offset[2] = -center.Z();
-        } else {
+			serializer->settings().transform.SetTranslation(center);
+			msg << "Centering model with offset (" << center.X() << "," << center.Y() << "," << center.Z() << ")";
+        } else if (model_offset) {
+			double offset[3];
             if (sscanf(offset_str.c_str(), "%lf;%lf;%lf", &offset[0], &offset[1], &offset[2]) != 3) {
                 std::cerr << "[Error] Invalid use of --model-offset\n";
                 delete serializer;
                 std::remove(output_temp_filename.c_str()); /**< @todo Windows Unicode support */
                 print_options(serializer_options);
                 return EXIT_FAILURE;
             }
-        }
-
-        std::stringstream msg;
-        msg << "Using model offset (" << offset[0] << "," << offset[1] << "," << offset[2] << ")";
+			msg << "Using model offset (" << offset[0] << "," << offset[1] << "," << offset[2] << ")";
+			serializer->settings().transform.SetTranslation(gp_XYZ(offset[0], offset[1], offset[2]));
+        } else {
+			IfcParse::IfcFile *ifc_file = context_iterator.getFile();
```

Comment:
Perhaps this can be moved to a method on `IfcGeom::Iterator`? Cleans up this file a bit and removes the need to expose the `getKernel()` method.